File: API/models/Dog.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const DogSchema = new Schema({
    name: {
        type: String,
        require: true,
        trim: true
    },
    breed:{
        type: String,
        require: true,
        trim: true
    },
    female: {
        type: Boolean, 
        require: true
    },
    weight: {
        type: Number,
        require: true,
        trim: true
    },
    color: {
        type: String,
        require: true,
        trim: true
    },
    birth:{
        type: Date,
        require: true
    },
    sterilized: {
        type: Boolean,
        require: true
    }
},
 {
    timestamps: true
});

DogSchema.methods = {
    
};


module.exports = mongoose.model("Dog", DogSchema);

File: LifeDogApp/app/src/main/java/com/carolinac/lifedogapp/data/entity/Dog.kt
package com.carolinac.lifedogapp.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity
data class Dog(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "dog_id")
    var dogId: Int,
    var name: String,
    var breed: String,
    var female: Boolean,
    var weight: Float,
    var color: String,
    var birth: Date?,
    var sterilized: Boolean,
    @ColumnInfo(name = "size_id")
    var sizeId: Int
)